sdk/src/prices/calculatePoolPrices.ts

```diff
@@ -0,0 +1,200 @@
+  TickArrayData,
+  TOKEN_MINTS,
+  WhirlpoolData,
+} from "../types/public";
+import { PoolUtil, SwapUtils } from "../utils/public";
+import { PDAUtil } from "../utils/public/pda-utils";
+
+export type PoolMap = { [key: string]: WhirlpoolData };
+export type TickArrayMap = { [key: string]: TickArrayData };
+export type PriceMap = { [key: string]: Decimal | null };
+export type TickSpacingAccumulator = null | { pool: WhirlpoolData; address: PublicKey };
+
+const USDC_THRESHOLD_AMOUNT = new BN(1_000_000_000);
+const PRICE_IMPACT_THRESHOLD = 1.05;
+
+// TODO: Auto-generate based on SOL price
+const SOL_THRESHOLD_AMOUNT = new BN(20_000_000_000);
+
+/**
+ * calculatePoolPrices will calculate the price of each token in the given mints array
+ * The price is calculated based on the pool with the highest liquidity
+ * In order for the pool to be considered, it must have sufficient liquidity
+ * Sufficient liquidity is defined by the thresholdAmount and priceImpactThreshold
+ * For example, if the thresholdAmount is 1000 USDC and the priceImpactThreshold is 0.01
+ * Then the pool must support 1000 USDC of liquidity without a price impact of 1%
+ * In order to calculate sufficient liquidity, the caller of the function must provide
+ * the tick arrays required to calculate the price impact
+ * @param mints
+ * @param poolMap
+ * @param tickArrayMap
+ * @returns PriceMap
+ */
+export function calculatePoolPrices(
+  mints: PublicKey[],
+  poolMap: PoolMap,
+  tickArrayMap: TickArrayMap
+): PriceMap {
+  mints = cleanMints(mints);
+
+  const prices = getPriceForQuoteToken(
+    mints.concat(new PublicKey(TOKEN_MINTS["SOL"])),
+    new PublicKey(TOKEN_MINTS["USDC"]),
+    poolMap,
+    tickArrayMap,
+    USDC_THRESHOLD_AMOUNT,
+    PRICE_IMPACT_THRESHOLD
+  );
+
+  const filteredMints = mints.filter((mint) => prices[mint.toBase58()] !== null);
+  const solMintSet = new Set(filteredMints.map((mint) => mint.toBase58()));
+  solMintSet.delete(TOKEN_MINTS["SOL"]);
+  solMintSet.delete(TOKEN_MINTS["USDC"]);
+
+  const solPrices = getPriceForQuoteToken(
+    mints,
+    new PublicKey(TOKEN_MINTS["SOL"]),
+    poolMap,
+    tickArrayMap,
+    SOL_THRESHOLD_AMOUNT,
+    PRICE_IMPACT_THRESHOLD
+  );
+
+  // Get SOL price, convert into USDC price
+  const solPrice = prices[TOKEN_MINTS["SOL"]];
+  if (!solPrice) {
+    return prices;
+  }
+
+  Object.entries(prices).forEach(([mint, price]) => {
+    if (price != null || solPrices[mint] === null) {
+      return;
+    }
+
+    const solPriceForMint = solPrices[mint];
+    if (!solPriceForMint) {
+      return;
+    }
+
+    prices[mint] = solPriceForMint.mul(solPrice);
+  });
+
+  return prices;
+}
+
+function liquidityThresholdCheck(
+  pool: WhirlpoolData,
+  tickArrays: TickArray[],
+  isTokenA: boolean,
+  amount: BN,
+  priceImpactThreshold: number
+): boolean {
+  const { estimatedAmountOut } = swapQuoteWithParams(
+    {
+      whirlpoolData: pool,
+      aToB: isTokenA,
+      amountSpecifiedIsInput: true,
```

Comment:
To confirm that X token can be swapped for 1000 USDC,  ExactOut (AmountIsInput=false) mode is helpful.

Liquidity of meme tokens tends to be deeper in the direction of swap USDC for meme token, but the reverse direction, liquidity is small.

This is just a feeling...
BaseToQuote should be ExactOut
QuoteToBase should be ExactIn